### ISSUE TITLE
pub get failed (1; So, because clash depends on both petitparser ^5.0.0 and pdf ^3.7.4, version solving failed.)

### ISSUE THREAD

**Issue by gcpbuild** [DavBfr/dart_pdf#1009]
**Describe the bug**
*I was* trying to add petitparser 5 as a dependency to my project
*I saw* pdf depends on xml version was restricted to less than 6 and versions of xml below 6 use versions of petitparser less than 5.
*I expected* there not to be an unnecessary upper bound to the xml as xml 6 will use petitparser 5

**To Reproduce**
Code snippet to reproduce the behavior:
pubspec
```dart
dependencies:
  flutter:
    sdk: flutter
  petitparser: ^5.0.0
  pdf: ^3.7.4

```

**Expected behaviour**
I expected to be able to use the latest petitparser version and not be held back.

**Screenshots**
```
$ flutter pub get
Because xml >=5.1.0 <5.2.0 depends on petitparser ^4.1.0 and xml >=5.2.0 <5.3.0 depends on petitparser ^4.2.0, xml >=5.1.0 <5.3.0 requires petitparser ^4.1.0.
And because xml >=5.3.0 <5.4.0 depends on petitparser ^4.3.0, xml >=5.1.0 <5.4.0 requires petitparser ^4.1.0.
And because xml >=5.4.0 depends on collection ^1.16.0 and every version of flutter from sdk depends on collection 1.15.0, if flutter from sdk and xml >=5.1.0 then petitparser ^4.1.0.
And because pdf >=3.7.1 depends on xml ^5.1.0 and clash depends on flutter from sdk, pdf >=3.7.1 requires petitparser ^4.1.0.
So, because clash depends on both petitparser ^5.0.0 and pdf ^3.7.4, version solving failed.
Running "flutter pub get" in clash...                                   
pub get failed (1; So, because clash depends on both petitparser ^5.0.0 and pdf ^3.7.4, version solving failed.)
```

**Flutter Doctor**

```
$ flutter doctor -v
[✓] Flutter (Channel stable, 2.10.4, on macOS 12.3.1 21E258 darwin-x64, locale en-GB)
    • Flutter version 2.10.4 at /Users/tommy/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c860cba910 (3 weeks ago), 2022-03-25 00:23:12 -0500
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/tommy/Library/Android/sdk
    • Platform android-31, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] IntelliJ IDEA Community Edition (version 2020.2.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 51.0.3
    • Dart plugin version 202.8070

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.3.1 21E258 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 100.0.4896.88

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

**Desktop (please complete the following information):**
 - [ ] iOS
 - [ ] Android
 - [X] Browser
 - [ ] Windows
 - [ ] Linux

**Smartphone (please complete the following information):**
 - Browser Chrome

**Additional context**
The xml package does have a version which uses the latest petitparser and an easy change to your pubspec would create a version of pdf I can use.  It's just a limitation of the flutter dependency system, I guess.

Thanks for your urgent attention.

**Comment by DavBfr** [comment]
The issue is on the package `xml`:

```
xml >=5.4.0 depends on collection ^1.16.0 and every version of flutter from sdk depends on collection 1.15.0, xml >=5.4.0 is incompatible with flutter from SDK
```

**Comment by gcpbuild** [comment]
Sorry if I didn't explain it well enough.  The package xml has been fixed and does use the new version of petitparser in its latest version 6.  The package pdf does not use the latest version of xml.  If you removed the < 6 restriction on xml in your pubspec I think the problem would go away. Thanks for answering so quickly.  The pdf package is so good it would be a shame to have to write my own.  Thanks.
 
```
dependencies:
  ...
  xml: ">=5.1.0 <6.0.0"
```

**Comment by DavBfr** [comment]
I'll change the dependencies. Can you confirm it fixes the issue when you change them locally?

**Comment by gcpbuild** [comment]
I cloned dart_pdf and made it a path dependency.  I set the dependency for xml to 6.0.1 but that didn't solve it.  It seems as though the image dependency also depends on xml 5 but there is no version of image which uses xml 6 yet.  

It might be quicker for me to clone petitparser and give it a different name so that the flutter dependency resolution doesn't complain, but that seems really really wrong.
